src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -1300,5 +1300,39 @@ public void Test_Issue657_enum()
                 ("BCP036", DiagnosticLevel.Error, "The property \"name\" expected a value of type \"'val1' | 'val2'\" but the provided value is of type \"'notAValidVal'\"."),
             });
         }
-    }
+
+        [TestMethod]
+        public void Test_Issue1985()
+        {
+            var result = CompilationHelper.Compile(@"
+resource aksDefaultPoolSubnet 'Microsoft.Network/virtualNetworks/subnets' existing = {
+  parent: virtualNetwork
+  name: aksDefaultPoolSubnetName
 }
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(aksDefaultPoolSubnet.id, 'Network Contributor')
+  scope: aksDefaultPoolSubnet
+  properties: {
+    principalId: aksServicePrincipalObjectId
+    roleDefinitionId: '4d97b98b-1d4f-4787-a291-c67834d212e7'
+  }
+  dependsOn: [
+    virtualNetwork
+    userAssignedIdentities
+  ]
+}
+");
+
+            result.Should().NotGenerateATemplate();
```

Comment:
Feels like `Should().NotGenerateATemplate()` should be implicit when `Should().HaveDiagnostics(...)` asserts on errors.